src/commands/mod.rs

```diff
@@ -1790,6 +1793,63 @@ fn edit_description(
     Ok(text_util::complete_newline(description))
 }
 
+fn edit_sparse(
+    workspace_root: &Path,
+    repo_path: &Path,
+    sparse: &[RepoPath],
+    settings: &UserSettings,
+) -> Result<Vec<RepoPath>, CommandError> {
+    let file = (|| -> Result<_, io::Error> {
+        let mut file = tempfile::Builder::new()
+            .prefix("editor-")
+            .suffix(".jjsparse")
+            .tempfile_in(repo_path)?;
+        for sparse_path in sparse {
+            file.write_all(
+                file_util::relative_path(workspace_root, &sparse_path.to_fs_path(workspace_root))
+                    .to_str()
+                    .unwrap()
```

Comment:
Ah, I inlined the definition of `WorkspaceCommandHelper::format_file_path()` (which uses `unwrap()`, and which is used by `sparse list`), but right it's better to avoid panics.

I changed to returning an error as you suggested, but we could alternatively write the path without any validation, pushing the problem onto the user as they edit the file.

Reply:
Sorry, I think I was thinking that these paths came from the user when I was reading this, but this is actually the code writing the paths to the file. I'm fine with `unwrap()` in that case. Feel free to switch back to that or leave as is.

Longer term, maybe there's a way to use the `camino` crate at least for workspace-root-relative paths like this. We have talked about using that in the past. I think what has prevented us from using it through the codebase is that it would prevent users from storing their repo in a path with a non-utf8 name.

Reply:
I didn't know about `camino`, it's interesting. Yeah I feel like it would be best to handle non-UTF-8 paths (especially since we're working with "boundaries" with the OS a lot, where it's fine to make files non-UTF-8 for instance). That said it's out of the scope of this PR.

I'll leave it as it is, as the error message is more descriptive about what failed (unlike unwrapping `None`).